lib/puppet/transaction/resource_harness.rb

```diff
@@ -158,7 +158,9 @@ def sync_if_needed(param, context)
       raise
     ensure
       if event
-        event.calculate_corrective_change(@persistence.get_system_value(context.resource.ref, param.name.to_s))
+        name = param.name.to_s
+        event.message = _("could not create change error message for %{name}") % { name: name } if event.message.nil?
```

Comment:
Probably more idiomatic to do something like this:

```suggestion
        event.message ||= _("could not create change error message for %{name}") % { name: name }
```